ui/src/components/RoleForm/index.jsx

```diff
@@ -200,7 +200,13 @@ export default class RoleForm extends Component {
                           key={scope}
                           button
                           component={Link}
-                          to={`/auth/scopes/${encodeURIComponent(scope)}`}
+                          to={
+                            scope.startsWith('assume:')
+                              ? `/auth/roles/${encodeURIComponent(
+                                  scope.replace('assume:', '')
+                                )}`
+                              : `/auth/scopes/${encodeURIComponent(scope)}`
+                          }
                           className={classes.listItemButton}>
```

Comment:
I think we would need to add the following diff in `ViewRole/index.jsx` to make sure the refresh happens:

```diff
diff --git a/ui/src/views/Roles/ViewRole/index.jsx b/ui/src/views/Roles/ViewRole/index.jsx
index 6080b45d6..3be7a534a 100644
--- a/ui/src/views/Roles/ViewRole/index.jsx
+++ b/ui/src/views/Roles/ViewRole/index.jsx
@@ -112,6 +112,7 @@ export default class ViewRole extends Component {
               {data && <ErrorPanel fixed error={data.error} />}
               {data && data.role && (
                 <RoleForm
+                  key={data.role.roleId}
                   role={data.role}
                   loading={loading}
                   onRoleDelete={this.handleDeleteRole}
```